app/src/main/java/org/fossasia/susi/ai/forgotpassword/ForgotPasswordPresenter.kt

```diff
@@ -0,0 +1,89 @@
+package org.fossasia.susi.ai.forgotpassword
+
+import android.content.Context
+import android.graphics.Color
+import android.util.Patterns
+import org.fossasia.susi.ai.R
+import org.fossasia.susi.ai.helper.Constant
+import org.fossasia.susi.ai.helper.CredentialHelper
+import org.fossasia.susi.ai.helper.PrefManager
+import org.fossasia.susi.ai.rest.ClientBuilder
+import org.fossasia.susi.ai.rest.responses.susi.ForgotPasswordResponse
+import retrofit2.Call
+import retrofit2.Callback
+import retrofit2.Response
+import java.net.UnknownHostException
+
+/**
+ * Created by meeera on 6/7/17.
+ */
+class ForgotPasswordPresenter : ForgotPasswordView {
+
+    var authResponseCall: Call<ForgotPasswordResponse>? = null
+
+    override fun cancelSignup() {
+        authResponseCall?.cancel()
+    }
+
+    override fun signup(email: String, url: String, isPersonalServerChecked: Boolean, context: Context, listener: ForgotPasswordView.onFinishListener) {
+        if (email.isEmpty()) {
```

Comment:
This all code should go in Interactor. You haven't even made an interactor. All the business logic should be in interactor/model and not in presenter.